File: integrations/mem0/python/scripts/deny_langchain.py
"""Test-only import tripwire for source and installed SDK checks."""
import sys
from importlib.abc import MetaPathFinder

class DenyLangChain(MetaPathFinder):
    def find_spec(self, fullname, path=None, target=None):
        if fullname.split(".")[0] in {"langchain", "langchain_core", "langchain_community", "langsmith"}:
            raise AssertionError("LangChain import attempted: " + fullname)

sys.meta_path.insert(0, DenyLangChain())

File: integrations/mem0/typescript/scripts/deny-langchain.mjs
/** Test-only import tripwire, also used against installed artifacts. */
export async function resolve(specifier, context, nextResolve) {
  if (/^(?:@langchain\/|langchain(?:\/|$)|langsmith(?:\/|$))/.test(specifier)) {
    throw new Error(`LangChain import attempted: ${specifier}`);
  }
  return nextResolve(specifier, context);
}
